### PR TITLE
Depth Of Tree

### PR DESCRIPTION
**Pull request by satyampgt4**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

The is program will give the depth of the binary tree

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1558

### CODE REVIEW

**Comment by satyampgt4** [comment]
I request @shraddhavp  @iamrajiv  @Neilblaze to review this PR. Thankyou

**Comment by satyampgt4** [comment]
@iamrajiv @atarax665 please review this PR. Thankyou

**Comment by shraddhavp** [comment]
please resolve the conflicts

File: C-Plus-Plus/graphs/Depth_of_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+
+    bin_tree t;
+    t.insert_data(50);
+    t.insert_data(40);
+    t.insert_data(60);
+    t.insert_data(30);
+    t.insert_data(45);
+    t.insert_data(55);
+    t.insert_data(70);
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/Depth_of_tree.cpp]: Don't push static input. Make it dynamic and let the User input the Data

File: C-Plus-Plus/graphs/Depth_of_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+        // depth of tree will max of depth of left subtree and depth of right subtree adding current node;
+        return  max(left_depth,rigth_depth)+1;
+    }
+
+    //depth of whole tree
+    int depth_of_tree()
+    {
+        return(depth(root));
+
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/Depth_of_tree.cpp]: Remove extra lines of Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Remove extra lines of Code. There are a lot of them!

**Comment by satyampgt4** [comment]
@HarshCasper i have done requested changes please check.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@satyampgt4 Add space and time complexity.

**Comment by satyampgt4** [comment]
@iamrajiv i have done requested changes please check.

**Review by iamrajiv — APPROVED** [review]

**Review by Neilblaze — APPROVED** [review]
LGTM :+1:

**Pull request merged**